### PR TITLE
feat(query): add optimize hints SET_VAR

### LINKED ISSUE #8833: Feature: support  SET_VAR hints
**Summary**

Description for this feature.

https://dev.mysql.com/doc/refman/8.0/en/optimizer-hints.html

```
SELECT /*+ SET_VAR(sort_buffer_size = 16M) */ name FROM people ORDER BY name;
INSERT /*+ SET_VAR(foreign_key_checks=OFF) */ INTO t2 VALUES(2);
SELECT /*+ SET_VAR(optimizer_switch = 'mrr_cost_based=off') */ 1;
```

### PR DESCRIPTION
**Pull request by TCeason**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

support optimize hints

The SET_VAR hint sets the session value of a system variable temporarily (for the duration of a single statement).

<div data-page-id="FDSfdYRfZojJBox6CZbcoZe5njh" data-docx-has-block-data="true"><div>

Hint Name | Description | Applicable Scopes
-- | -- | --
SET_VAR | Set variable during statement execution | Global

</div></div><span data-lark-record-data="{&quot;isCut&quot;:false,&quot;rootId&quot;:&quot;FDSfdYRfZojJBox6CZbcoZe5njh&quot;,&quot;parentId&quot;:&quot;FDSfdYRfZojJBox6CZbcoZe5njh&quot;,&quot;blockIds&quot;:[49],&quot;recordIds&quot;:[&quot;GmowdaKiIoiO40x8lVlcH6Wpn9C&quot;],&quot;recordMap&quot;:{&quot;GmowdaKiIoiO40x8lVlcH6Wpn9C&quot;:{&quot;id&quot;:&quot;GmowdaKiIoiO40x8lVlcH6Wpn9C&quot;,&quot;snapshot&quot;:{&quot;type&quot;:&quot;table&quot;,&quot;parent_id&quot;:&quot;FDSfdYRfZojJBox6CZbcoZe5njh&quot;,&quot;comments&quot;:[],&quot;locked&quot;:false,&quot;hidden&quot;:false,&quot;author&quot;:&quot;7049617019452653570&quot;,&quot;columns_id&quot;:[&quot;col54074a3f-1e2b-43fe-a7c9-8801143e9a7d&quot;,&quot;cole5b8be7d-90d3-4bef-bb14-9194d09fa9d1&quot;,&quot;col53a2dc8d-a2df-4c69-a510-66a308567c9d&quot;],&quot;rows_id&quot;:[&quot;row53c70e25-cdeb-4727-8848-bd729536d2ed&quot;,&quot;row06cbcd7c-6c07-403a-a2e8-de76828091c7&quot;],&quot;column_set&quot;:{&quot;col53a2dc8d-a2df-4c69-a510-66a308567c9d&quot;:{&quot;column_width&quot;:192},&quot;col54074a3f-1e2b-43fe-a7c9-8801143e9a7d&quot;:{&quot;column_width&quot;:141},&quot;cole5b8be7d-90d3-4bef-bb14-9194d09fa9d1&quot;:{&quot;column_width&quot;:291}},&quot;cell_set&quot;:{&quot;row06cbcd7c-6c07-403a-a2e8-de76828091c7col53a2dc8d-a2df-4c69-a510-66a308567c9d&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;R84cdCKACooAi4xEDtac7khGnzd&quot;},&quot;row06cbcd7c-6c07-403a-a2e8-de76828091c7col54074a3f-1e2b-43fe-a7c9-8801143e9a7d&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;Ke8Ed0UECoASKUx4sf4cVVqWnTd&quot;},&quot;row06cbcd7c-6c07-403a-a2e8-de76828091c7cole5b8be7d-90d3-4bef-bb14-9194d09fa9d1&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;Pqeodw4SWoucwwx23qvcdLz3nZf&quot;},&quot;row53c70e25-cdeb-4727-8848-bd729536d2edcol53a2dc8d-a2df-4c69-a510-66a308567c9d&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;NSAedIuMioWMMUxuwaqcNHxknQb&quot;},&quot;row53c70e25-cdeb-4727-8848-bd729536d2edcol54074a3f-1e2b-43fe-a7c9-8801143e9a7d&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;POcodqWcUo06G0xoZbocu1f5ngc&quot;},&quot;row53c70e25-cdeb-4727-8848-bd729536d2edcole5b8be7d-90d3-4bef-bb14-9194d09fa9d1&quot;:{&quot;merge_info&quot;:{&quot;row_span&quot;:1,&quot;col_span&quot;:1},&quot;block_id&quot;:&quot;RwOUd4c8goKUwyxS2p4cfWlgngb&quot;}},&quot;header_row&quot;:true}},&quot;POcodqWcUo06G0xoZbocu1f5ngc&quot;:{&quot;id&quot;:&quot;POcodqWcUo06G0xoZbocu1f5ngc&quot;,&quot;snapshot&quot;:{&quot;type&quot;:&quot;table_cell&quot;,&quot;parent_id&quot;:&quot;GmowdaKiIoiO40x8lVlcH6Wpn9C&quot;,&quot;comments&quot;:[],&quot;locked&quot;:false,&quot;hidden&quot;:false,&quot;author&quot;:&quot;7049617019452653570&quot;,&quot;children&quot;:[&quot;IMk2d82IKo08AOxBpyrcQs3EnwB&quot;,&quot;TKyEduU0KoiIaixstHIc83upnng&quot;]}},&quot;IMk2d82IKo08AOxBpyrcQs3EnwB&quot;:{&quot;id&quot;:&quot;IMk2d82IKo08AOxBpyrcQs3EnwB&quot;,&quot;snapshot&quot;:{&quot;type&quot;:&quot;text&quot;,&quot;parent_id&quot;:&quot;POcodqWcUo06G0xoZbocu1f5ngc&quot;,&quot;comments&quot;:[],&quot;locked&quot;:false,&quot;hidden&quot;:false,&quot;author&quot;:&quot;7049617019452653570&quot;,&quot;children&quot;:[],&quot;text&quot;:{&quot;apool&quot;:{&quot;nextNum&quot;:2,&quot;numToAttrib&quot;:{&quot;0&quot;:[&quot;author&quot;,&quot;7049617019452653570&quot;],&quot;1&quot;:[&quot;bold&quot;,&quot;true&quot;]}},&quot;initialAttributedTexts&quot;:{&quot;attribs&quot;:{&quot;0&quot;:&quot;*0*1+9&quot;},&quot;text&quot;:{&quot;0&quot;:&quot;Hint Name&quot;}}},&quot;align&quot;:&quot;center&quot;,&quot;folded&quot;:false}},&quot;TKyEduU0KoiIaixstHIc83upnng&quot;:{&quot;id&quot;:&quot;TKyEduU0KoiIaixstHIc83upnng&quot;,&quot;snapshot&quot;:{&quot;type&quot;:&quot;text&quot;,&quot;parent_id&quot;:&quot;POcodqWcUo06G0xoZbocu1f5ngc&quot;,&quot;comments&quot;:[],&quot;locked&quot;:false,&quot;hidden&quot;:false,&quot;author&quot;:&quot;7049617019452653570&quot;,&quot;children&quot;:[],&quot;text&quot;:{&quot;apool&quot;:{&quot;nextNum&quot;:0,&quot;numToAttrib&quot;:{}},&quot;initialAttributedTexts&quot;:{&quot;attribs&quot;:{&quot;0&quot;:&quot;&quot;},&quot;text&quot;:{&quot;0&quot;:&quot;&quot;}}},&quot;align&quot;:&quot;center&quot;,&quot;folded&quot;:false}},&quot;RwOUd4c8goKUwyxS2p4cfWlgngb&quot;:{&quot;id&quot;:&quot;RwOUd4c8goKUwyxS2p4cfWlgngb&quot;,&quot;snapshot&quot;:{&quot;type&quot;:&quot;table_cell&quot;,&quot;parent_id&quot;:&quot;GmowdaKiIoiO40x8lVlcH6Wpn9C&quot;,&quot;comments&quot;:[],&quot;locked&quot;:false,&quot;hidden&quot;:false,&quot;author&quot;:&quot;7049617019452653570&quot;,&quot;children&quot;:[&quot;VCGSdcCYwo8EooxQDE8ciF0Wnzc&quot;]}},&quot;VCGSdcCYwo8EooxQDE8ciF0Wnzc&quot;:{&quot;id&quot;:&quot;VCGSdcCYwo8EooxQDE8ciF0Wnzc&quot;,&quot;snapshot&quot;:{&quot;type&quot;:&quot;text&quot;,&quot;parent_id&quot;:&quot;RwOUd4c8goKUwyxS2p4cfWlgngb&quot;,&quot;comments&quot;:[],&quot;locked&quot;:false,&quot;hidden&quot;:false,&quot;author&quot;:&quot;7049617019452653570&quot;,&quot;children&quot;:[],&quot;text&quot;:{&quot;apool&quot;:{&quot;nextNum&quot;:2,&quot;numToAttrib&quot;:{&quot;0&quot;:[&quot;author&quot;,&quot;7049617019452653570&quot;],&quot;1&quot;:[&quot;bold&quot;,&quot;true&quot;]}},&quot;initialAttributedTexts&quot;:{&quot;attribs&quot;:{&quot;0&quot;:&quot;*0*1+b&quot;},&quot;text&quot;:{&quot;0&quot;:&quot;Description&quot;}}},&quot;align&quot;:&quot;center&quot;,&quot;folded&quot;:false}},&quot;NSAedIuMioWMMUxuwaqcNHxknQb&quot;:{&quot;id&quot;:&quot;NSAedIuMioWMMUxuwaqcNHxknQb&quot;,&quot;snapshot&quot;:{&quot;type&quot;:&quot;table_cell&quot;,&quot;parent_id&quot;:&quot;GmowdaKiIoiO40x8lVlcH6Wpn9C&quot;,&quot;comments&quot;:[],&quot;locked&quot;:false,&quot;hidden&quot;:false,&quot;author&quot;:&quot;7049617019452653570&quot;,&quot;children&quot;:[&quot;B2yIdCeksocISUxMlKxcJBkyn7d&quot;]}},&quot;B2yIdCeksocISUxMlKxcJBkyn7d&quot;:{&quot;id&quot;:&quot;B2yIdCeksocISUxMlKxcJBkyn7d&quot;,&quot;snapshot&quot;:{&quot;type&quot;:&quot;text&quot;,&quot;parent_id&quot;:&quot;NSAedIuMioWMMUxuwaqcNHxknQb&quot;,&quot;comments&quot;:[],&quot;locked&quot;:false,&quot;hidden&quot;:false,&quot;author&quot;:&quot;7049617019452653570&quot;,&quot;children&quot;:[],&quot;text&quot;:{&quot;apool&quot;:{&quot;nextNum&quot;:2,&quot;numToAttrib&quot;:{&quot;0&quot;:[&quot;author&quot;,&quot;7049617019452653570&quot;],&quot;1&quot;:[&quot;bold&quot;,&quot;true&quot;]}},&quot;initialAttributedTexts&quot;:{&quot;attribs&quot;:{&quot;0&quot;:&quot;*0*1+h&quot;},&quot;text&quot;:{&quot;0&quot;:&quot;Applicable Scopes&quot;}}},&quot;align&quot;:&quot;center&quot;,&quot;folded&quot;:false}},&quot;Ke8Ed0UECoASKUx4sf4cVVqWnTd&quot;:{&quot;id&quot;:&quot;Ke8Ed0UECoASKUx4sf4cVVqWnTd&quot;,&quot;snapshot&quot;:{&quot;type&quot;:&quot;table_cell&quot;,&quot;parent_id&quot;:&quot;GmowdaKiIoiO40x8lVlcH6Wpn9C&quot;,&quot;comments&quot;:[],&quot;locked&quot;:false,&quot;hidden&quot;:false,&quot;author&quot;:&quot;7049617019452653570&quot;,&quot;children&quot;:[&quot;DasQdWyq2ocmMgxMTVfcqgcEnnh&quot;]}},&quot;DasQdWyq2ocmMgxMTVfcqgcEnnh&quot;:{&quot;id&quot;:&quot;DasQdWyq2ocmMgxMTVfcqgcEnnh&quot;,&quot;snapshot&quot;:{&quot;type&quot;:&quot;text&quot;,&quot;parent_id&quot;:&quot;Ke8Ed0UECoASKUx4sf4cVVqWnTd&quot;,&quot;comments&quot;:[],&quot;locked&quot;:false,&quot;hidden&quot;:false,&quot;author&quot;:&quot;7049617019452653570&quot;,&quot;children&quot;:[],&quot;text&quot;:{&quot;apool&quot;:{&quot;nextNum&quot;:3,&quot;numToAttrib&quot;:{&quot;0&quot;:[&quot;author&quot;,&quot;7049617019452653570&quot;],&quot;1&quot;:[&quot;bold&quot;,&quot;true&quot;],&quot;2&quot;:[&quot;link&quot;,&quot;https%3A%2F%2Fdev.mysql.com%2Fdoc%2Frefman%2F8.0%2Fen%2Foptimizer-hints.html%23optimizer-hints-set-var&quot;]}},&quot;initialAttributedTexts&quot;:{&quot;attribs&quot;:{&quot;0&quot;:&quot;*0*1*2+7&quot;},&quot;text&quot;:{&quot;0&quot;:&quot;SET_VAR&quot;}}},&quot;align&quot;:&quot;center&quot;,&quot;folded&quot;:false}},&quot;Pqeodw4SWoucwwx23qvcdLz3nZf&quot;:{&quot;id&quot;:&quot;Pqeodw4SWoucwwx23qvcdLz3nZf&quot;,&quot;snapshot&quot;:{&quot;type&quot;:&quot;table_cell&quot;,&quot;parent_id&quot;:&quot;GmowdaKiIoiO40x8lVlcH6Wpn9C&quot;,&quot;comments&quot;:[],&quot;locked&quot;:false,&quot;hidden&quot;:false,&quot;author&quot;:&quot;7049617019452653570&quot;,&quot;children&quot;:[&quot;FqoEdys6mo40ioxwjjnckyXanNg&quot;]}},&quot;FqoEdys6mo40ioxwjjnckyXanNg&quot;:{&quot;id&quot;:&quot;FqoEdys6mo40ioxwjjnckyXanNg&quot;,&quot;snapshot&quot;:{&quot;type&quot;:&quot;text&quot;,&quot;parent_id&quot;:&quot;Pqeodw4SWoucwwx23qvcdLz3nZf&quot;,&quot;comments&quot;:[],&quot;locked&quot;:false,&quot;hidden&quot;:false,&quot;author&quot;:&quot;7049617019452653570&quot;,&quot;children&quot;:[],&quot;text&quot;:{&quot;apool&quot;:{&quot;nextNum&quot;:1,&quot;numToAttrib&quot;:{&quot;0&quot;:[&quot;author&quot;,&quot;7049617019452653570&quot;]}},&quot;initialAttributedTexts&quot;:{&quot;attribs&quot;:{&quot;0&quot;:&quot;*0+13&quot;},&quot;text&quot;:{&quot;0&quot;:&quot;Set variable during statement execution&quot;}}},&quot;align&quot;:&quot;center&quot;,&quot;folded&quot;:false}},&quot;R84cdCKACooAi4xEDtac7khGnzd&quot;:{&quot;id&quot;:&quot;R84cdCKACooAi4xEDtac7khGnzd&quot;,&quot;snapshot&quot;:{&quot;type&quot;:&quot;table_cell&quot;,&quot;parent_id&quot;:&quot;GmowdaKiIoiO40x8lVlcH6Wpn9C&quot;,&quot;comments&quot;:[],&quot;locked&quot;:false,&quot;hidden&quot;:false,&quot;author&quot;:&quot;7049617019452653570&quot;,&quot;children&quot;:[&quot;VwgCd6I6koWqSaxQHyVcnR6Wn8g&quot;]}},&quot;VwgCd6I6koWqSaxQHyVcnR6Wn8g&quot;:{&quot;id&quot;:&quot;VwgCd6I6koWqSaxQHyVcnR6Wn8g&quot;,&quot;snapshot&quot;:{&quot;type&quot;:&quot;text&quot;,&quot;parent_id&quot;:&quot;R84cdCKACooAi4xEDtac7khGnzd&quot;,&quot;comments&quot;:[],&quot;locked&quot;:false,&quot;hidden&quot;:false,&quot;author&quot;:&quot;7049617019452653570&quot;,&quot;children&quot;:[],&quot;text&quot;:{&quot;apool&quot;:{&quot;nextNum&quot;:1,&quot;numToAttrib&quot;:{&quot;0&quot;:[&quot;author&quot;,&quot;7049617019452653570&quot;]}},&quot;initialAttributedTexts&quot;:{&quot;attribs&quot;:{&quot;0&quot;:&quot;*0+6&quot;},&quot;text&quot;:{&quot;0&quot;:&quot;Global&quot;}}},&quot;align&quot;:&quot;center&quot;,&quot;folded&quot;:false}},&quot;FDSfdYRfZojJBox6CZbcoZe5njh&quot;:{&quot;id&quot;:&quot;FDSfdYRfZojJBox6CZbcoZe5njh&quot;,&quot;snapshot&quot;:{&quot;type&quot;:&quot;page&quot;,&quot;parent_id&quot;:&quot;&quot;,&quot;comments&quot;:null,&quot;locked&quot;:false,&quot;hidden&quot;:false,&quot;author&quot;:&quot;7049617019452653570&quot;,&quot;children&quot;:[&quot;Bq2EdOygQo4CuCxUhk7cYiSZn3g&quot;,&quot;HYGYdIyUYoYseyx2nU3cvTmfnEb&quot;,&quot;KUGAdgEGmo4cSex2HblcKfdUnme&quot;,&quot;VsSUdieEgoGcMUxULtEcpA5RnFh&quot;,&quot;OiYkda6Wuo8s6sxgVkCc4akTnbg&quot;,&quot;AKM4dIowqoC0ouxwx4Icr0B3nIe&quot;,&quot;Pa0Gd0SAMo08k4xuUVZcghdxnOc&quot;,&quot;Hi4wdkqQyoaucCxk2SIcQ3htnjP&quot;,&quot;JeA0dYYi4owuyUxoXr9cHrv9nNc&quot;,&quot;ECCcdOWIEoisQWxcXEmcebEkntc&quot;,&quot;X0uUd8EEAomWaIxUJKdcdxLPnEg&quot;,&quot;XycUdO2O8oKgQyxsTAQcOQPEnqe&quot;,&quot;AuKsdMA8Womom2x8KcUcnHL3nrc&quot;,&quot;BqmKdAYMSow4aExI3p4csbSKnrd&quot;,&quot;Wu2Ods0QCoc08IxqwFecX5SdnWg&quot;,&quot;JEIsdIUGMo8cMexKORKcyVFjnId&quot;,&quot;VQUmdO0ycoUUa6x8Qd7cuZlrnYb&quot;,&quot;GwsedwOkkoC6QUxsdDHc0eshnad&quot;,&quot;HGUgdkgyso4Yo0xuY4rcpvuZnwd&quot;,&quot;Lgi2de4I2oAmAAxywX6c1iSGn4c&quot;,&quot;HWMEdweYyoeeI2xGWyOcux96nEf&quot;,&quot;LqoYdCyyioiC48xoNi2cjjCXnCd&quot;,&quot;C6QQdge0yoC06GxAlRwc1svInLh&quot;,&quot;T84Md2IeuoAWWoxoFqQcytn2nVg&quot;,&quot;S6Asd8CgIoU6MexqsBLcvmJonPf&quot;,&quot;TaWwdYiCMoi6Q4xmixlcqbX1nVf&quot;,&quot;WqssdUsgUogOUYx6hvTcif42ncc&quot;,&quot;N4I4dMKWaocyCcx6gSccr0E6nPf&quot;,&quot;GuiCdEemmoMyi0xExmDcQseenWc&quot;,&quot;DmW4d0Gmoo8cAox6R5IclgcQnXb&quot;,&quot;BIUsdg2W4oeSKCxK2cmcp427nrh&quot;,&quot;TYo2dIkQ2oWwMAxSaPAcFGjtnbd&quot;,&quot;JQm8dIEA4oYkqGxcPErcnBLfnPh&quot;,&quot;T0WGdEcGEo2GO4xWqe8caXSfnFh&quot;,&quot;PIWYdmWaMoCw8Exg9bcchYawnPg&quot;,&quot;MAK2dQCauo2o0wxDVfvcvkD7nlV&quot;,&quot;DCWqdgU0koYwgQxQv0KcWUehnBf&quot;,&quot;UUMidmIeYoY0i0x6lJFcGXwrnGd&quot;,&quot;U6mKdo2YwoMY6axIXsNcFk1knrc&quot;,&quot;IaKYdCI0IoiWAExIPWucq0pznXd&quot;,&quot;AA0cdyqeko6eSKx0mp9cM4SHn2e&quot;,&quot;BCOMdWoS2o28Cgxa6lvcJFqHnUd&quot;,&quot;R6oudaIU2o0oymx8Pvjcrq97nbf&quot;,&quot;Wissd42QgoUMgqxMByKcUgWUnUg&quot;,&quot;HS8idMaqsoqoIExmE6hcN0YPnKe&quot;,&quot;TC6Ed22QQo2USkxucCscRW0VnSf&quot;,&quot;GUGwdKI8UoqEUAxgKi2cKAs1ngb&quot;,&quot;GmowdaKiIoiO40x8lVlcH6Wpn9C&quot;,&quot;XOuwdIAquoykeCxQ9Y1c0yQEnJc&quot;,&quot;RGmIdo6IuoSqyWxyCG4cCWLHnZf&quot;,&quot;Niysde2Coo8cKsxIJD2cfsndnrh&quot;,&quot;BOUCdmOu2oia2axeOZicCwCEneb&quot;,&quot;CYkAdSM64o02oIxWyofcK128nUg&quot;,&quot;VWSQdoEoQooyGexua9zc2FuBnoe&quot;,&quot;B2K8dS8EOooisixiGercFwE9n8g&quot;,&quot;HYi2dqGoyo02SOx6ISpcQyOlnHf&quot;,&quot;J4kgdsOAgo4SW2x0QMHcVPvjnkh&quot;,&quot;Om4OdUycQoa6sWxyIPScOUXMnsb&quot;,&quot;Lu2odkeogooumsxAF7ec2zeXn5d&quot;,&quot;FY6edGQYUocCk2x0AM1c1GN4nah&quot;,&quot;OQi8dKAiYo4A88xW0cScbvsvnHc&quot;,&quot;Pq64dGoCsosOMsxgjnQc9aDknGb&quot;,&quot;VIowd8U62o4S0gxe03Pcj3m9nWg&quot;,&quot;FAu4dw8SSo4OwSxozZ3ceH6cnoh&quot;,&quot;UsQ4dGeMOooymcxiSAjcHKfVnRh&quot;,&quot;XGOydawIgoi8u6xkHCLclOlYn2f&quot;,&quot;R8Oudo2KWoIaEUxGUBEcD5SPnOh&quot;,&quot;EwCAd0iOYoGcGAx2xU2cqDulnVc&quot;,&quot;SigAdk8A2oUsAIxo1n3c3VLYnoe&quot;,&quot;OA2odu6AyoYEAQxc1FUc3ZVyn2b&quot;,&quot;AWMcdQ0QmoO2SqxMFYAcHNPQnvg&quot;,&quot;Js6edkUeEoSwaAxikzZcBA6xnnf&quot;,&quot;X080daI0wowiQ4xqcvOcoDVMnjh&quot;,&quot;TowmdaKIko4I06xUBQ6cHjZFn2e&quot;,&quot;DAkKdUOWOogKKoxYtlUcSXXhnEb&quot;,&quot;OiuydwYgMocUSgxg7Jrc3lsgntb&quot;,&quot;XCocdKYiQosMaWxUBkncAmqvnQ0&quot;,&quot;Qi2UduyCsoweMAxCcMqcWYZvnOg&quot;,&quot;W2gCdWiiIo2OksxMLhuclziYnZe&quot;,&quot;AiomdiokMouGesxCIq1c4wGpnSf&quot;,&quot;JKSMdk0euoM6MaxeWLrc7tt8nRo&quot;,&quot;MYWQdySWgoy4y8xUdwOc6O54nye&quot;,&quot;MEEqdQCEgo60WIxInAgc5LPonvb&quot;,&quot;Ui8MdkAUcoKcgwxwJR2ck0hwnAb&quot;,&quot;VOEydOy0WokoucxySkhcvX1knce&quot;,&quot;D8c2dAGUWoKeQYxUXpjcUPAqnbh&quot;,&quot;FUYudQqsAoE2aExi2pvcSgJfn5e&quot;,&quot;BOm2dcC0CoqyYexAxzIc4SrMnTf&quot;,&quot;XOM8dmgoWoYwSix8MplcdbjdnBp&quot;,&quot;HkuQdcqwgoIua8xYZJPcXoBIn1c&quot;,&quot;HO2gdIamuoQaSCxyQLlcl9vJnVb&quot;,&quot;SSyGdSIaIou2KYxU9rtcYuA3nnh&quot;,&quot;KUOWdYwmqoI0cuxMW5Qcfh2ensb&quot;,&quot;I2G4dc2qqoGE8YxeFFrcq7tFnNP&quot;,&quot;KCiedEo8uoiI4yx4lZ2c9GY3nSh&quot;,&quot;MGSEdugemoMmAixiiz3ccv66nIh&quot;,&quot;QWoAdKueIoe8CKxeL6ucRJBenRm&quot;,&quot;MaKGdcqg0ooAkqx8r1ycIuJrnCg&quot;,&quot;NUMKdgMkyo60iMxk5JIcMwlBnDa&quot;,&quot;DywKdiUiQoOgMcxohUJcgNzcnCe&quot;,&quot;Bsy8dq2q0oOIwUxARZgcz8xunVh&quot;,&quot;KucYdiY8OossG0xg1v1cPiJZncd&quot;,&quot;WmC4d2Cw4oKa2axsLibcYOxtnSb&quot;,&quot;WYU2dA6gkowugYxmwAwcGLHTnhd&quot;,&quot;N8qqd6OeuoyUE0xYOGNcnvWsnlb&quot;,&quot;M0SidEIocoImKKxe4nHcxx8Wnug&quot;,&quot;IiQgdAeKMoW8CQxeqXEcOgx7nHd&quot;,&quot;C4gsdQKwYoigsgxIJRdcvB1unfe&quot;,&quot;ZqSSd8ccGoGc68xQzK7cPvaynIc&quot;,&quot;NmkqdSgSAouc48xCQufcLj3jnoe&quot;,&quot;XEYOd6ykAo64AWxUDbDcLQ9LnTf&quot;,&quot;ZaUqdGqeCo2mCuxU8z2cwC63nTO&quot;,&quot;XuQ4d8MWCoWi4Mx690wc7Txhnrd&quot;,&quot;SSgydWE2Koieeix62QfcNgZNnPf&quot;,&quot;QseWdEgsoowS2ExUrJDcuiR6nMd&quot;,&quot;SgQkdaEO4oWwu6xmgMHcDVd8nAe&quot;,&quot;MAgSdMEico4Cmyxodcacqk27nTh&quot;,&quot;K8mAdwkaUoiAsAxEvnHcXSZSnsd&quot;,&quot;XIyId84omouCmkxsNZgcaSsjnbg&quot;,&quot;CIqcdgwM8oQG0SxEX3nc2moinHf&quot;],&quot;text&quot;:{&quot;apool&quot;:{&quot;nextNum&quot;:1,&quot;numToAttrib&quot;:{&quot;0&quot;:[&quot;author&quot;,&quot;7049617019452653570&quot;]}},&quot;initialAttributedTexts&quot;:{&quot;attribs&quot;:{&quot;0&quot;:&quot;*0+f&quot;},&quot;text&quot;:{&quot;0&quot;:&quot;Optimizer Hints&quot;}}},&quot;align&quot;:&quot;&quot;}}},&quot;payloadMap&quot;:{&quot;IMk2d82IKo08AOxBpyrcQs3EnwB&quot;:{&quot;level&quot;:1},&quot;TKyEduU0KoiIaixstHIc83upnng&quot;:{&quot;level&quot;:1},&quot;VCGSdcCYwo8EooxQDE8ciF0Wnzc&quot;:{&quot;level&quot;:1},&quot;B2yIdCeksocISUxMlKxcJBkyn7d&quot;:{&quot;level&quot;:1},&quot;DasQdWyq2ocmMgxMTVfcqgcEnnh&quot;:{&quot;level&quot;:1},&quot;FqoEdys6mo40ioxwjjnckyXanNg&quot;:{&quot;level&quot;:1},&quot;VwgCd6I6koWqSaxQHyVcnR6Wn8g&quot;:{&quot;level&quot;:1}},&quot;extra&quot;:{&quot;mention_page_title&quot;:{},&quot;external_mention_url&quot;:{}},&quot;isKeepQuoteContainer&quot;:false,&quot;selection&quot;:[{&quot;id&quot;:49,&quot;type&quot;:&quot;block&quot;,&quot;recordId&quot;:&quot;GmowdaKiIoiO40x8lVlcH6Wpn9C&quot;}],&quot;pasteFlag&quot;:&quot;f5c2f87b-031d-4d95-a486-a15c4b35298c&quot;}" data-lark-record-format="docx/record" class="lark-record-clipboard"></span>

### Syntax

SET_VAR syntax permits setting only a single variable, but multiple hints can be given to set multiple variables:

/*+ SET_VAR(key=value) SET_VAR(key=value) ... */

### Usage

* The hint syntax must follow immediately after an `INSERT`, `UPDATE`, `DELETE`, `SELECT`, or `REPLACE` keyword that begins the statement block.
* If contains the same key, only the first key will be valid.
* If one key parse or binder failed, all hints will be invalid.

Closes #8833

### CODE REVIEW

File: tests/suites/0_stateless/05_hints/05_0001_set_var.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+echo "select /*+ SET_VAR(timezone=x) SET_VAR(timezone=select 1) */ name, value from system.settings where name='timezone';" |  $MYSQL_CLIENT_CONNECT
+echo "select /*+ SET_VAR(storage_read_buffer_size=200) SET_VAR(timezone=x) */ name, value from system.settings where name='timezone' or name = 'storage_read_buffer_size';" |  $MYSQL_CLIENT_CONNECT
+
+echo "drop database if exists set_var;" | $MYSQL_CLIENT_CONNECT
+echo "create database set_var;" | $MYSQL_CLIENT_CONNECT
+echo "create table set_var.test(id int);" | $MYSQL_CLIENT_CONNECT
+echo "insert /*+SET_VAR(timezone='Asia/Shanghai') SET_VAR(storage_read_buffer_size=200)*/ into set_var.test values(1)" | $MYSQL_CLIENT_CONNECT
+echo "insert /*+SET_VAR(timezone='Asia/Shanghai') (storage_read_buffer_size=200)*/ into set_var.test values(3)" | $MYSQL_CLIENT_CONNECT
+echo "select /*+SET_VAR(timezone='Asia/Shanghai') SET_VAR(storage_read_buffer_size=200)*/ * from set_var.test order by id" | $MYSQL_CLIENT_CONNECT
```
Comment (sundy-li) [inline tests/suites/0_stateless/05_hints/05_0001_set_var.sh:20]: could add case:
select /*+SET_VAR(timezone='Asia/Shanghai') */ * from system.settings where name = 'timezone';

File: src/query/ast/src/ast/statements/hint.rs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "/*+ ")?;
+        for hint in &self.hints_list {
+            write!(f, "SET_VAR(")?;
+            write!(f, "{}", hint.name)?;
+            write!(f, "=")?;
+            write!(f, "{}", hint.expr)?;
+            write!(f, ") ")?;
+        }
+        write!(f, " */")
```
Comment (andylokandy) [inline src/query/ast/src/ast/statements/hint.rs:42]:
```suggestion
        write!(f, "*/")
```

File: src/query/ast/src/ast/statements/statement.rs [hunk 3]
```diff
@@ -272,9 +273,12 @@ impl Display for Statement {
             Statement::Delete {
                 table_reference,
                 selection,
-                ..
+                hints,
             } => {
                 write!(f, "DELETE FROM {table_reference}")?;
+                if let Some(hints) = hints {
+                    write!(f, "{}", hints)?;
```
Comment (andylokandy) [inline src/query/ast/src/ast/statements/statement.rs:280]:
```suggestion
                    write!(f, "{} ", hints)?;
```

File: src/query/ast/src/parser/statement.rs [hunk 4]
```diff
@@ -1322,6 +1327,35 @@ pub fn unset_source(i: Input) -> IResult<UnSetSource> {
...
+        rule! {
+            SET_VAR ~ ^"(" ~ ^#ident ~ ^"=" ~ #subexpr(0) ~ ^")"
+        },
+        |(_, _, name, _, expr, _)| HintItem { name, expr },
+    )(i)
+}
+
+pub fn any_token(i: Input) -> IResult<&[Token]> {
+    Ok((i.slice(1..), i.0.slice(1..)))
+}
```
Comment (andylokandy) [inline src/query/ast/src/parser/statement.rs:1341]: it's better to move it to `src/query/ast/src/util.rs`
```suggestion
pub fn any_token(i: Input) -> IResult<&Token> {
    move |i| match i.0.get(0).filter(|token| token.kind != TokenKind::EOI) {
        Some(token) => Ok((i.slice(1..), token)),
        _ => Err(nom::Err::Error(Error::from_error_kind(
            i,
            ErrorKind::Other("expected any token but reached the end"),
        ))),
    }
}
```

File: src/query/ast/src/parser/statement.rs [hunk 5]
```diff
@@ -1322,6 +1327,35 @@ pub fn unset_source(i: Input) -> IResult<UnSetSource> {
...
+pub fn hint(i: Input) -> IResult<Hint> {
+    let hint = map(
+        rule! {
+            "/*+" ~ #set_var_hints+ ~ "*/"
+        },
+        |(_, hints_list, _)| Hint { hints_list },
+    );
+    let invalid_hint = map(
+        rule! {
+            "/*+" ~ ^((!"*/" ~ #any_token)*) ~ "*/"
```
Comment (andylokandy) [inline src/query/ast/src/parser/statement.rs:1352]:
```suggestion
            "/*+" ~ (!"*/" ~ #any_token)* ~ "*/"
```

File: src/query/ast/tests/it/testdata/lexer.txt [hunk 6]
```diff
@@ -28,6 +28,48 @@ x'deadbeef' -- a hex string\n 'a string literal\n escape quote by '' or \\\'. '
...
+---------- Input ----------
+select /* x /*+ yy */ */ 1
+---------- Output ---------
+[(SELECT, "select", 0..6), (HintSuffix, "*/", 22..24), (LiteralInteger, "1", 25..26), (EOI, "", 26..26)]
+
+
+---------- Input ----------
+select 1 + /*+ foo
+---------- Output ---------
+[(SELECT, "select", 0..6), (LiteralInteger, "1", 7..8), (Plus, "+", 9..10)]
```
Comment (TCeason) [inline src/query/ast/tests/it/testdata/lexer.txt:46]: This test need end with EOI

**Pull request merged**